docs/configuration.asciidoc

```diff
@@ -0,0 +1,265 @@
+[[configuration]]
+== Configuration
+
+[float]
+=== Gradle configuration
+
+Configure your application at compile time within your application's `build.gradle` file:
+
+[source,groovy]
+----
+// Android app's build.gradle file
+plugins {
+    //...
+    id "co.elastic.apm.android" version "[latest_version]"
+}
+
+elasticApm {
+    // Minimal configuration
+    serverUrl = "https://your.elastic.server"
+
+    // Optional
+    serviceName = "your app name" <1>
+    serviceVersion = "0.0.0" <2>
+    secretToken = "your server auth token" <3>
+}
+----
+<1> Defaults to your `android.defaultConfig.applicationId` value.
+<2> Defaults to your `android.defaultConfig.versionName` value.
+<3> Defaults to null.
+
+All of the values provided in the Gradle configuration can be overridden with the following environment variables:
+
+|===
+|Config |Associated Environment variable
+
+|serviceName
+|`ELASTIC_APM_SERVICE_NAME`
+
+|serviceVersion
+|`ELASTIC_APM_SERVICE_VERSION`
+
+|serverUrl
+|`ELASTIC_APM_SERVER_URL`
+
+|secretToken
+|`ELASTIC_APM_SECRET_TOKEN`
+
+|===
+
+[float]
+=== Runtime configuration
+
+The runtime configuration is provided within your https://developer.android.com/reference/android/app/Application[Application] class at the time of initializing the Elastic agent.
+This configuration can override any previously-set compile time configuration.
```

Comment:
This _can_ override compile-time config? Or it _does_ overwrite compile-time config when set?

Reply:
Nice catch, it _does_ overwrite it, compile config stands no chance against the runtime one 😄